nft_ingester/src/bin/backfill/main.rs

```diff
@@ -0,0 +1,384 @@
+    if let Some(slots_str) = args.slots {
+        info!("Processing specific slots provided via command line.");
+        for part in slots_str.split(',') {
+            let slot_str = part.trim();
+            if let Ok(slot) = slot_str.parse::<u64>() {
+                slots_to_process.push(slot);
+            } else {
+                warn!("Invalid slot number provided: {}", slot_str);
+            }
+        }
+
+        // Remove duplicates and sort slots
+        let mut slots_set = HashSet::new();
+        slots_to_process = slots_to_process
+            .into_iter()
+            .filter(|x| slots_set.insert(*x))
+            .collect();
+        slots_to_process.sort_unstable();
+
+        if slots_to_process.is_empty() {
+            error!("No valid slots to process. Exiting.");
+            return;
+        }
+
+        info!("Total slots to process: {}", slots_to_process.len());
+    }
+
+    // Set up progress bar
+    let total_slots = if !slots_to_process.is_empty() {
+        slots_to_process.len() as u64
+    } else {
+        // Get the last slot
+        let mut iter = source_db.raw_iterator_cf(&source_db.cf_handle(RawBlock::NAME).unwrap());
+        iter.seek_to_last();
+        if !iter.valid() {
+            error!("Failed to seek to last slot");
+            return;
+        }
+        let last_slot = iter
+            .key()
+            .map(|k| u64::from_be_bytes(k.try_into().expect("Failed to decode the last slot key")))
+            .expect("Failed to get the last slot");
+
+        // Determine the starting slot
+        let start_slot = if let Some(start_slot) = args.start_slot {
+            info!("Starting from slot: {}", start_slot);
+            start_slot
+        } else {
+            iter.seek_to_first();
+            iter.key()
+                .map(|k| {
+                    u64::from_be_bytes(k.try_into().expect("Failed to decode the start slot key"))
+                })
+                .expect("Failed to get the start slot")
+        };
+
+        info!("Start slot: {}, Last slot: {}", start_slot, last_slot);
+        last_slot - start_slot + 1
+    };
+
+    let progress_bar = Arc::new(ProgressBar::new(total_slots));
+    progress_bar.set_style(
+        ProgressStyle::default_bar()
+            .template(
+                "{spinner:.green} [{elapsed_precise}] [{bar:40.cyan/blue}] {percent}% \
+                 ({pos}/{len}) {msg}",
+            )
+            .expect("Failed to set progress bar style")
+            .progress_chars("#>-"),
+    );
+
+    // Spawn worker tasks
+    let mut worker_handles = Vec::new();
+    for _ in 0..num_workers {
+        let consumer = consumer.clone();
+        let progress_bar = progress_bar.clone();
+        let slots_processed = slots_processed.clone();
+        let rate = rate.clone();
+        let shutdown_token = shutdown_token.clone();
+
+        let slot_receiver = slot_receiver.clone();
+
+        let handle = tokio::spawn(async move {
```

Comment:
if we have multiple handles, then it might be more idiomatic to use `JoinSet`. Not critical, though